### PR TITLE
karavictl validations for input flags when creating and updating resources

### PR DESCRIPTION
**Pull request by sharmilarama**
# Description

This PR updates all create/update commands for tenant/role/rolebinding/storage to check for valid, non-empty inputs and display appropriate error messages if input is not valid.

# Issues

List the issues impacted by this PR:

| Issue ID |
| -------- |
| #78 |

# Checklist:

- [x] I have performed a self-review of my own changes.

### CODE REVIEW

**Comment by ioeicicdsvc** [comment]
@sharmilarama: Thank you for submitting your pull request!

Please make sure you have followed our contributing guidelines. We will review it as soon as possible.

------------------------------------------------------------------------------------------------------------------------
We want your feedback! If you have any questions or suggestions regarding our contributing process/workflow, please reach out to us at karavi@dell.com.

File: cmd/karavictl/cmd/role_create.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ var roleCreateCmd = &cobra.Command{
 		var rff roles.JSON
 		for _, v := range roleFlags {
 			t := strings.Split(v, "=")
+			if len(t) < 5 {
```
Comment (tdawe) [inline cmd/karavictl/cmd/role_create.go:50]: Could use a const here for the value of 5 to improve code readability

**Review by gallacher — APPROVED** [review]

File: cmd/karavictl/cmd/role_create.go (hunk 1 above)
Comment (sharmilarama) [reply cmd/karavictl/cmd/role_create.go]: fixed

File: cmd/karavictl/cmd/storage_update.go [hunk 2]
```diff
@@ -61,15 +74,37 @@ var storageUpdateCmd = &cobra.Command{
 			Password string
 			Insecure bool
 		}{
-			Type:     flagStringValue(cmd.Flags().GetString("type")),
-			Endpoint: flagStringValue(cmd.Flags().GetString("endpoint")),
-			SystemID: flagStringValue(cmd.Flags().GetString("system-id")),
-			User:     flagStringValue(cmd.Flags().GetString("user")),
+			Type:     verifyInput("type"),
```
[the remark this answers is not in the document]
Comment (sharmilarama) [reply cmd/karavictl/cmd/storage_update.go:77]: I will add this for required flags, but this does not check for empty strings, so I will keep the other check also to catch empty values.

File: cmd/karavictl/cmd/storage_update.go [hunk 3]
```diff
@@ -37,20 +38,32 @@ var storageUpdateCmd = &cobra.Command{
...
 			return v
 		}
 		flagBoolValue := func(v bool, err error) bool {
 			if err != nil {
-				log.Fatal(err)
+				errAndExit(err)
 			}
 			return v
 		}
+		verifyInput := func(Type string) string {
```
[the remark this answers is not in the document]
Comment (sharmilarama) [reply cmd/karavictl/cmd/storage_update.go]: fixed.

**Comment by medegw01** [comment]
e2e pass

**Review by ianbibby — APPROVED** [review]
👍

**Review by alikdell — APPROVED** [review]

**Pull request merged**